configstore/tests/test_zintegration.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+import configstore
+
+DOTENV_CONTENTS = u'''
+SECRET_KEY=1234dot
+ENVIRONMENT=PRODUCTION
+'''
+
+DOTENV_EMPTY = u''''''
+
+
+def test_env_var_dotenv_success(monkeypatch, tmpdir):
+    monkeypatch.setenv('ENVIRONMENT', 'STAGING')
+    p = tmpdir.join("config.env")
+    p.write(DOTENV_CONTENTS)
+
+    store = configstore.Store([
+        configstore.EnvVarBackend(),
+        configstore.DotenvBackend(str(p)),
+    ])
+
+    secret_key = store.get_setting('SECRET_KEY')
+    environment = store.get_setting('ENVIRONMENT')
+
+    assert secret_key == '1234dot'
+    assert environment == 'STAGING'
+
+
+def test_dotenv_env_var_success(monkeypatch, tmpdir):
+    monkeypatch.setenv('ENVIRONMENT', 'STAGING')
+    p = tmpdir.join("config.env")
+    p.write(DOTENV_CONTENTS)
+
+    store = configstore.Store([
+        configstore.DotenvBackend(str(p)),
+        configstore.EnvVarBackend(),
```

Comment:
Comme la seule chose qui change de ce test c'est l'ordre des backends dans le constructeur du store. Tu pourrais probablement paramétrer le test pour l'écrire une seule fois, mais qu'il soit rouler avec les deux tuples.

[pytest parametrizing fixtures and test functions](https://docs.pytest.org/en/latest/parametrize.html)

Reply:
Non, le résultat pour `get_setting('ENVIRONMENT')` est différent justement.

Ce test est utile si plus tard on veut définir par exemple un ordre de priorité par backend (envvar toujours override dotenv etc) pour que l’ordre des instances backends dans le store soit ignoré.

Pour ce cas, garder deux tests simples et presque identiques me semble très bien.

Reply:
Bah, ça pourrait quand même se paramétrer, mais si tu crois que c'est mieux de garder deux tests distinct, ça me va ! :)